Introduce Chapter-3 Outro Page

shaavan opened the pull request:
Resolves #502

<img width="1496" alt="Screenshot 2023-06-25 at 3 25 23 PM" src="https://github.com/saving-satoshi/saving-satoshi/assets/85434418/31af9522-46a3-47ae-8f5b-ef45a1451855">

vinayaksh42 commented:
Now we will be able to address this issue: https://github.com/saving-satoshi/saving-satoshi/issues/480 once this pull request is merged.

vinayaksh42 commented:
Overall looks great for desktop view. Spotted a few issues related to mobile view:
- Text overflow: https://github.com/saving-satoshi/saving-satoshi/issues/524
- In the Figma mockup the background image is reduced in size and completely displayed, followed by the text. Where as in the preview of this PR the image is displayed differently.
![IMG_3671](https://github.com/saving-satoshi/saving-satoshi/assets/54861487/0c4d085b-d357-4ed5-b231-22ea88454bbc)

<img width="235" alt="Screenshot 2023-06-26 at 7 17 49 PM" src="https://github.com/saving-satoshi/saving-satoshi/assets/54861487/ed8a6b2e-fb66-4d49-83d1-2a463ff06782">

benalleng commented on content/lessons/chapter-3/outro-1.tsx:
```diff
@@ -0,0 +1,35 @@
+  title: 'chapter_three.outro_one.title',
+  image: '/assets/images/chapter-3-finish.jpg',
+  theme: 'bg-transparent/20',
+  secondaryTheme: 'bg-[#362042]',
```
Lets go ahead and change the secondaryTheme to this color `bg-[#3c5c6e]`,  found from a dropper within the picture

shaavan replied:
Addressed in e9c7b5d0c9ba853b0efb8b13594477225a469bd1
Thanks for the suggestion!

benalleng commented:
> In the Figma mockup the background image is reduced in size and completely displayed, followed by the text. Where as in the preview of this PR the image is displayed differently.

I assumed that was a mistake from an early design phase

shaavan commented on content/lessons/chapter-3/outro-1.tsx:
```diff
@@ -0,0 +1,35 @@
+  return (
+    <ChapterEnd
+      image={metadata.image}
+      className="relative -mt-[63px] flex h-[1150px] overflow-y-scroll md:-mt-[70px] md:h-screen"
```
We should definitely address the issue with the fixed height.

In addition, we can also take the opportunity to make changes to the chapter-2/outro-1 section. To keep the focus of this pull request specific, we can overlook this minor issue and ensure consistency in the outro pages of both chapter-2 and chapter-3.

benalleng replied:
You can use your discretion but it's set at `h-[992px]` in the figma I think that will make it a bit easier to see on most mobile phones as this particular outro image is not as busy as chapter-2 it feels empty at this height

afterburn replied:
I definitely don't think we should be setting a hard pixel value for the height. I'm having some weird issues here (notice the scrollbar):

![image](https://github.com/saving-satoshi/saving-satoshi/assets/10757768/77c822b4-39eb-4dea-9f40-475e62bf757c)

We should set this to `h-full` or `h-screen`, if we want to position the image differently we should use the appropriate CSS properties for that (object-position, background-position, etc).

shaavan commented:
@vinayaksh42 

> In the Figma mockup the background image is reduced in size and completely displayed, followed by the text. Whereas in the preview of this PR the image is displayed differently.

Great catch!

I also noticed the same discrepancy between the Figma designs and the implementation in the outro pages of other chapters. To ensure we address this issue effectively, let's create a separate pull request dedicated to resolving it. This approach will help us keep the scope of the current pull request well-defined and manageable.

benalleng commented on content/lessons/chapter-2/outro-1.tsx:
```diff
@@ -7,7 +7,7 @@ export const metadata = {
   image: '/assets/images/chapter-2-finish.jpg',
   theme: 'bg-transparent/20',
-  secondaryTheme: 'bg-[#362042]',
+  secondaryTheme: 'bg-[#3c5c6e]',
```
Looks like you accidentally made the color change for chapter-2

shaavan replied:
Oops, sorry! 😅
Let's fix it! ✨

shaavan replied:
Fixed it in 625aa12a29bad54faa7d722a979c0d427abd53fb
Thanks for catching it!

benalleng reviewed (changes requested):
Thanks! Small mistake but I think after this it will be good to go, we can address some design changes like the height when we have a chance later on

shaavan commented:
Updated from e9c7b5d0c9ba853b0efb8b13594477225a469bd1 to 625aa12a29bad54faa7d722a979c0d427abd53fb ([pr531.02](https://github.com/shaavan/saving-satoshi/commits/pr531.02) -> [pr531.03](https://github.com/shaavan/saving-satoshi/commits/pr531.03), [diff](https://github.com/shaavan/saving-satoshi/compare/pr531.02..pr531.03))
Addressed @benalleng suggestion

- Fixed a mistake. Change the `secondaryTheme` of `chapter-3/outro-1` instead of `chapter-2/intro-1`.

ecurrencyhodler commented:
Can confirm navbar overview is no longer an issue on mobile. And address the background image reduction by adding it to #480.

![image](https://github.com/saving-satoshi/saving-satoshi/assets/32662508/488be743-001a-4913-8837-01b9645cedd8)

afterburn commented on content/lessons/chapter-3/index.ts:
```diff
@@ -1,11 +1,13 @@
 import * as Intro1 from './intro-1'
 import * as Intro2 from './intro-2'
 import * as SOL1 from './solo-1'
+import * as OUT1 from './outro-1'
```
Nitpick but I'm not sure why we are using this different naming convention here.

shaavan replied:
I missed the naming convention followed in other chapters while writing this part.
Thanks for catching it!

Addressed in 01cf4fc113094935cbe2b2dec8741ef5c82409e3

afterburn reviewed (changes requested):
Thanks @shaavan, requested some changes

shaavan commented:
Updated from 625aa12a29bad54faa7d722a979c0d427abd53fb to 01cf4fc113094935cbe2b2dec8741ef5c82409e3 ([pr531.03](https://github.com/shaavan/saving-satoshi/commits/pr531.03) -> [pr531.04](https://github.com/shaavan/saving-satoshi/commits/pr531.04), [diff](https://github.com/shaavan/saving-satoshi/compare/pr531.03..pr531.04))
Addressed @afterburn comment

- Followed naming convention for lesson and outro in `index.ts`.
